### DISCUSSION TITLE
Designing a Database Schema Similar to Udemy

### DISCUSSION THREAD

**Discussion by emiliobiasi** [pocketbase/pocketbase#5382] [Q&A]
```json
[
    {
        "name": "contents",
        "schema": [
            {"name": "title", "type": "text"},
            {"name": "module", "type": "relation"},
            {"name": "description", "type": "text"},
            {"name": "file", "type": "file"},
            {"name": "upload_id", "type": "text"},
            {"name": "playback_id", "type": "text"},
            {"name": "users_completed", "type": "relation"}
        ]
    },
    {
        "name": "courses",
        "schema": [
            {"name": "title", "type": "text"},
            {"name": "description", "type": "text"},
            {"name": "by", "type": "relation"},
            {"name": "club", "type": "relation"},
            {"name": "image", "type": "file"},
            {"name": "price", "type": "number"},
        ]
    },
    {
        "name": "modules",
        "schema": [
            {"name": "title", "type": "text"},
            {"name": "course", "type": "relation"}
        ]
    },
    {
        "name": "purchased_courses",
        "schema": [
            {"name": "user", "type": "relation"},
            {"name": "course", "type": "relation"},
            {"name": "completed", "type": "bool"},
            {"name": "certificate", "type": "file"},
        ]
    }
]
```

I have these four collections in the database and am currently working on tracking the progress of users through their courses. My goal is to keep track of X contents completed out of Y total contents, both at the course level and the module level. I'm looking for the best way to structure this for efficiency and accuracy.

Does anyone have any ideas or has done something similar? The way I've structured it seems like I'll need to make repeated calls to PocketBase, or if I use relation/back-relation, I might hit some limit. I'm not sure if the way I did it would work.

**Comment by guilhermeguidetti** [comment]
[pb_schema.json](https://github.com/user-attachments/files/16753056/pb_schema.json)
DB-Schema ^^

I structured my query as follows:
```js
await pb
      .collection("purchased_courses")
      .getFullList({
        filter: `user="${userId}"`,
        expand:
          "course.modules(course).contents(module).user_progress(content)",
      });
```

While this approach seems reliable, I'm not entirely sure if it's the best one. The documentation mentions a limit for back-relations, but it's unclear to me whether I'm exceeding that limit.

**Reply by guilhermeguidetti** [reply]
@ganigeorgiev do you know if it's possible to retrieve user_progress based on a specific user? For example, user_progress(user). How would this work in this case? Currently, it returns the progress of all users in the content, but I only want the progress filtered by the user.
